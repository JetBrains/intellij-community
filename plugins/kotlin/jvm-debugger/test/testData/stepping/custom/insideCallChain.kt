package insideCallChain

fun String.convert(function: (String) -> String) = function("convert: $this")
fun foo() = "three"
fun String.convert2(function: (String) -> String) = function("convert: $this")
fun foo2() = "three"

fun main() {
    "one".convert { "two $it" }
        // SMART_STEP_INTO_BY_INDEX: 1
        // RESUME: 1
        //Breakpoint!, lambdaOrdinal = -1
        .convert { foo() }
        .convert2 { foo2() }

    "one".convert { "two $it" }
        // SMART_STEP_INTO_BY_INDEX: 2
        // STEP_INTO: 1
        // RESUME: 1
        //Breakpoint!, lambdaOrdinal = -1
        .convert { foo() }
        .convert2 { foo2() }

    "one".convert { "two $it" }
        // SMART_STEP_INTO_BY_INDEX: 3
        // RESUME: 1
        //Breakpoint!, lambdaOrdinal = -1
        .convert { foo() }
        .convert2 { foo2() }

    "one".convert { "two $it" }
        // SMART_STEP_INTO_BY_INDEX: 4
        // STEP_INTO: 1
        // RESUME: 1
        //Breakpoint!, lambdaOrdinal = -1
        .convert { foo() }
        .convert2 { foo2() }
}

// IGNORE_K2
