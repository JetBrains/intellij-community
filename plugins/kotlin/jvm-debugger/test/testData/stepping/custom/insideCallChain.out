LineBreakpoint created at insideCallChain.kt:13 lambdaOrdinal = -1
LineBreakpoint created at insideCallChain.kt:21 lambdaOrdinal = -1
LineBreakpoint created at insideCallChain.kt:28 lambdaOrdinal = -1
LineBreakpoint created at insideCallChain.kt:36 lambdaOrdinal = -1
Run Java
Connected to the target VM
insideCallChain.kt:13
insideCallChain.kt:3
insideCallChain.kt:21
insideCallChain.kt:21
insideCallChain.kt:4
insideCallChain.kt:28
insideCallChain.kt:5
insideCallChain.kt:36
insideCallChain.kt:37
insideCallChain.kt:6
Disconnected from the target VM

Process finished with exit code 0
